lib/datadog/profiling/component.rb

```diff
@@ -110,6 +116,52 @@ def self.build_profiler_component(settings:, agent_settings:, optional_tracer:)
         end
       end
 
+      private_class_method def self.get_allocation_sample_every(settings)
+        allocation_sample_rate = settings.profiling.advanced.experimental_allocation_sample_rate
+
+        if allocation_sample_rate <= 0
+          raise("Allocation sample rate must be a positive integer. Was #{allocation_sample_rate}")
+        end
+
+        allocation_sample_rate
+      end
+
+      private_class_method def self.enable_allocation_profiling?(settings, allocation_sample_every)
+        allocation_profiling_enabled = settings.profiling.advanced.experimental_allocation_enabled
+
+        if allocation_profiling_enabled
+          Datadog.logger.warn(
+            "Enabled experimental allocation profiling: allocation_sample_rate=#{allocation_sample_every}. This is " \
+            'experimental, not recommended, and will increase overhead!'
+          )
+        end
+
+        if allocation_profiling_enabled && !Ext.allocation_sampling_supported?
+          Datadog.logger.warn(
+            "Current Ruby version (#{RUBY_VERSION}) does not officially support allocation profiling but it was " \
+            'requested. There may be unexpected problems during execution.'
+          )
+        end
```

Comment:
I'm thinking: the list of issues with the tracepoint basically could be grouped into

a) It breaks if you're using Ractors
b) It breaks if you're on Ruby 3.2.0 to 3.2.2

I know this is not what we do today, but what do you think of maybe:

1. Never allowing the feature to be enabled on Ruby 3.2.0 to 3.2.2, even if enabled. "Sorry, not supported, please use 3.1.x/3.2.3 or 3.3.0+"? (Caveat: 3.2.3 is actually not out, but the backport is in the branch, so we'd be excluding the latest stable Ruby for now but 3.3 will be out in December so maybe that's ok-ish for the state of this feature?)
2. Then we're free to be more exact in this error message for a) and say that due to VM bugs, if you're using this feature together with Ractors, expect issues, including crashes.